domain/credential/state/state.go

```diff
@@ -44,27 +44,38 @@ func credentialKeyMap(key corecredential.Key) sqlair.M {
 	}
 }
 
-func (st *State) credentialUUID(ctx context.Context, tx *sqlair.TX, key corecredential.Key) (string, error) {
+// credentialID finds and returns the uuid for the cloud credential identified
+// by key. If no credential is found then an error of
+// [credentialerrors.NotFound] is returned.
+func (st *State) credentialID(ctx context.Context, tx *sqlair.TX, key corecredential.Key) (corecredential.ID, error) {
 	selectQ := `
 SELECT &M.uuid
 FROM v_cloud_credential
 WHERE name = $M.credential_name
 AND owner_name = $M.owner
 AND cloud_name = $M.cloud_name
 `
-
 	selectStmt, err := st.Prepare(selectQ, sqlair.M{})
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 	uuid := sqlair.M{}
 	err = tx.Query(ctx, selectStmt, credentialKeyMap(key)).Get(&uuid)
 	if errors.Is(err, sql.ErrNoRows) {
-		return "", fmt.Errorf("cloud credential %q %w%w", key, errors.NotFound, errors.Hide(err))
+		return "", fmt.Errorf("cloud credential %q %w", key, credentialerrors.NotFound)
 	} else if err != nil {
 		return "", fmt.Errorf("fetching cloud credential %q: %w", key, err)
 	}
-	return uuid["uuid"].(string), nil
+
+	uuidStr, exists := uuid["uuid"]
```

Comment:
Simon asked for it a few months ago in another PR. Hence why I am being safe.

Reply:
In a different conversation, given all this is because sqlair doesn't yet support scalars, Gustavo has indicated we should avoid using maps and instead use small bespoke structs. Then the issue above goes away.
Not for this PR as such, just for next time etc.

Reply:
@wallyworld is correct